### PR TITLE
Added compression option

### PR DESCRIPTION
**Pull request by assertchris**
download 1

### CODE REVIEW

**Comment by Ocramius** [comment]
no-unit-tests-hasta-la-vista-baby

File: src/SerializableClosure.php [hunk 1]
```diff
@@ -107,8 +128,22 @@ public function bindTo($newthis, $newscope = 'static')
...
-            return serialize($this->data);
+            $this->data = $data = $this->data ?: $this->serializer->getData($this->closure, true);
+
+            if ($this->compressClosure) {
+                $file = tempnam('', '');
+                $code = $data['code'];
+
+                file_put_contents($file, '<?php ' . $code);
+
+                exec("php -w {$file}", $output);
```
Comment (Ocramius) [inline src/SerializableClosure.php]: You really wanna gzip rather than doing this sort of simplified whitespace compression :X

Comment (assertchris) [inline src/SerializableClosure.php]: Yes!

Comment (jwpage) [inline src/SerializableClosure.php]: Could use http://php.net/manual/en/function.php-strip-whitespace.php here instead?

**Comment by assertchris** [comment]
@jeremeamia I've added a test for compression. Is it behaviour you think fits in Super Closure, or should we close this?

**Comment by aztech-dev** [comment]
Would it not be possible to move all the (de)compression logic in a specific Serializer (that decorates the standard serializer) ? because moar SRP :)

**Comment by jeremeamia** [comment]
I don't feel like this is a feature that needs to be added to SuperClosure. Why can't you just compress the result of `$serializer->serialize($closure))`? Adding this makes the serializer unnecessarily mutable.

**Comment by jeremeamia** [comment]
But thank you @assertchris for the idea, and sorry the reply took so long. Craziness of PNWPHP and my new job are dying down now. Time to catch up.

**Pull request closed without merging**